DAQ_System/Source/Adapter/DataLogger/SdDataLogger.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+* SD Data Logger
+* Authors:        Cristian Cruz
+*                 Ethan Fischer
+*
+* Email:          cris14.alex@gmail.com
+*                 ejf180001@utdallas.edu
+* 
+* (c) 2022 Dallas Formula Racing - Embedded Firmware Team
+* Formula SAE International Collegiate Chapter
+* GPL-3.0 License
+*/
+
+#include "SdDataLogger.hpp"
+#include "FATFileSystem.h"
+#include "SDBlockDevice.h"
+
+namespace adapter {
+
+// used to set a new file name for each new log session
+int log_session = 0;
+char file_name[16] = "\0";
+
+// block device and file system declarations
+SDBlockDevice block_device;
+FATFileSystem file_system;
```

Comment:
Can we make these private variables?

Reply:
these are already private because they're under 'private:' in the header file, doing "private char file_name[16] = '\0\;" results in an expected unqualified-id syntax error.

Reply:
Ahh I see. In that case, I think it would be a great idea to remove them (these are variables/objects declared in the global scope of the namespace — outside of the class).

Hmm.. here's what I've noticed:
* `file_name_` in the data logging class is not being used
* `file_name` in this snippet is not being used
* `file_name` in main.cpp as passed into `FileOpen()` as the argument for `file_path`

I was thinking that maybe we make the file name a private variable with a setter-method. But now that I think about it, you really only need the file name/path when using `FileOpen()` — the class should only care about the FILE *. 

The way it's currently set up, the data logging class  requires the exact same FILE * for three its methods. I am starting to think that we should keep the `file_name` variable in main.cpp, and make a private FILE * for the class. This FILE * could be initialized within `FileOpen()`

If only I had this epiphany earlier... I introduced double-pointers for nothing lol

If you agree with this, should this be handled in this PR, or made as a separate, quick re-factor task?

Reply:
Lemme just make the quick change that I think you might be getting at and let me know what you think, if it's more than that we could probably make it a separate refactor task.